src/emu/luaprinter.h

```diff
@@ -0,0 +1,116 @@
+// copyright-holder: Golden Child
+/***************************************************************************
+
+    luaprinter.h
+
+    Utility class for printing functions
+
+***************************************************************************/
+
+#ifndef MAME_LUAPRINTER_H
+#define MAME_LUAPRINTER_H
+
+#pragma once
+
+#include "emu.h"
+#include "png.h"
+#include "emuopts.h"
+
+class luaprinter {
+
+public:
+	luaprinter(device_t& thisdevice, std::string dummyname);
+	~luaprinter() { lp_removefromprinterlist(); }
```

Comment:
If this is supposed to be a mixin that integrates with the `device_t` lifecycle, it should implement `device_interface`.  Also, passing a string by value like this is usually not what you want to do.